apps/src/applab/applab.js

```diff
@@ -854,33 +866,53 @@ Applab.render = function() {
   );
 };
 
-Applab.exportApp = function(expoOpts) {
-  Applab.runButtonClick();
-  var html = document.getElementById('divApplab').outerHTML;
-  studioApp().resetButtonClick();
-
-  // TODO: find another way to get this info that doesn't rely on globals.
+/**
+ * Export the project for web or use within Expo.
+ * @param {Object} expoOpts
+ */
+Applab.exportApp = async function(expoOpts) {
   const appName = project.getCurrentName() || 'my-app';
 
-  const {mode, expoSnackId, iconUri, splashImageUri} = expoOpts || {};
-  if (mode === 'expoGenerateApk') {
-    return generateExpoApk(
-      {
+  const {mode} = expoOpts || {};
+  switch (mode) {
```

Comment:
Rather than having a switch both here and in expoInteractWithApk, it might make more sense to split expoInteractWithApk into 3 different functions and call them individually. It will also be more reusable that way.

Reply:
Yes, this stems from how we share a single `exportApp` function for both `AdvancedShareOptions` and `ExportDialog`. This master function branches into different functions for different parts of the export process. When we remove/refactor the export code into one UX flow, we can convert this to bind separate functions for each phase (instead of binding `exportApp` as a property through redux's `pageConstants`, we can bind several functions like `expoInteractWithApk()`)

Reply:
I think it makes sense to take a step in that direction now. Having the multiple case statements with shared constant strings worries me. i.e. it opens us to issues any time we add or edit a case statement. I'd prefer to have that in one place rather than 3. On that note, we use these strings across multiple files. This seems like also a good time to move them to constants.